Realm/RLMUtil.mm

```diff
@@ -248,6 +249,50 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
     return outArray;
 };
 
+NSArray *RLMCollectionValueForKey(NSString *key, RLMRealm *realm, RLMObjectSchema *objectSchema, size_t count, size_t (^indexGenerator)(size_t)) {
+    if (count == 0) {
+        return @[];
+    }
+    BOOL keyIsSelf = [key isEqualToString:@"self"];
+    NSMutableArray *results = [NSMutableArray arrayWithCapacity:count];
+    RLMObjectBase *accessor;
+    tightdb::Table *table = nullptr;
+    if (!keyIsSelf) {
```

Comment:
I don't think it requires any duplicate logic other than a loop, and seem cleaner to me:
```objc
NSMutableArray *results = [NSMutableArray arrayWithCapacity:count];
 if ([key isEqualToString:@"self"]) {      
    for (size_t i = 0; i < count; i++) {
        size_t rowIndex = indexGenerator(i);
        [results addObject:RLMCreateObjectAccessor(realm, objectSchema, rowIndex) ?: NSNull.null];
    }
    return results;
}

RLMObjectBase *accessor = [[objectSchema.accessorClass alloc] initWithRealm:realm schema:objectSchema];    
tightdb::Table *table = objectSchema.table;
for (size_t i = 0; i < count; i++) {
    size_t rowIndex = indexGenerator(i);
    accessor->_row = (*table)[rowIndex];
    RLMInitializeSwiftListAccessor(accessor);
    [results addObject:[accessor valueForKey:key] ?: NSNull.null];
}
```